web-console/knikubevirt/NoStorageClassAvailable/NoStorageClassAvailable.md

```diff
@@ -0,0 +1,32 @@
+# Storage class isn’t defined
+
+Currently in our designs we have a default storage class used when a disk is added in the Create VM flow, but we do not cover the absence of a default storage class.
+This design covers the state of no storage class selected and determines how the user should be notified.
+
+During the process of creating the VM, in the Storage step, the user needs to add a disk.
+
+![Add disk action](img/Add-disk-default.png)
+
+Clicking the ‘Add disk’ button will open a modal, where the user is asked to fill in the disk’s details.
+
+We want to allow the user to set the Storage class within the wizard, with no need to be taken out of it.
+
+![Fill in disk details](img/Add-disk-modal-details-required.png)
+
+Clicking the ‘Add disk’ button at the bottom will add the disk to the list view, but the Storage class will be shown in the disks table as ‘Non available’.
+
+![Empty Storage class popover](img/Disks-list.png)
+
+Clicking on the ‘non available’ Storage class will show a popover with more info and an action link which will take the user to the ‘Create Storage class’ modal.
+
+In the VM list view, the status of the VM would still be pending untill the user would create a storage class and mark it as default.
+
+Clicking on the ‘pending’ status of the VM will show a popover with more info and an action link which will take the user to the ‘Create Storage class’ modal.
+
+![vmlist with popover](img/vmlist.png)
+
+'Create Storage class’ modal
+
+![Create storage class modal](img/Create-storage-class-modal.png)
```

Comment:
Wouldn't we be able to do this via a modal or do we have to send them to the storage class page and have them do it via the in page form? http://openshift.github.io/openshift-origin-design/web-console/4.0-designs/storage/storage
Also, I don't see a method to mark a storage class as default in the UI, @jelkosz can we add that? What happens if you mark one as default and then mark another as default? Can we remove it from the first?
This might be outside of the scope of this PR :)

Reply:
tjelinek@redhat.com Would it be "too much" to ask to develop the modal when we already have the in page form? I think the advantage of keeping the modal is that the user wouldn't feel detached from the environment they already are, while if we send them to create the SC via the in-page form, they might feel they are a bit 'lost'.

Reply:
@yfrimanm I spoke to @jelkosz about this. He suggested we direct the user to the storage class section of the ui and allow them to create a new one there. I think that makes sense. We don't want to rebuild something like this flow for an edge case and I think it's acceptable for the user to go to the SC section to create a new one. I've seen that there is a PR up for the default storage class work too so that will tie in nicely with what you've done here. #271